### ISSUE TITLE
Meerkat Authentication plan

### ISSUE THREAD

**Issue by davekempe** [meerkat-dashboard/meerkat#31]
Meerkat needs authentication for some purposes, while still being able to support viewable URLs easily for monitors on walls etc.
The current plan is as follows:
* Single admin user for editing maps configured in config file, plain text is OK
* Userlogin is prompted and enforced for home page (list of diagrams) 
* Each diagram has a token generated for it which is appended automatically in the view list on the front page.
* The view links are only viewable with the generated token appended to the URL.
* Editing is only available via logging into the web UI with the admin username and password.
* In the advanced view of a diagram, you should be able to see the token, and regenerate it if required.

This along with having an IcingaAPI user with read only rights, should provide a decent balance of security and usability. Yes its possible to leak/share read only versions of the maps with whoever can reach the meerkat server, however given that meerkat has no ability to make changes this should be acceptable. We can expand on this model later on if we need more fine grained control.

**Comment by ollytom** [comment]
From @sol1-matt: 

It has been requested that Meerkat be able to have view only users and admin user.

Meerkat has no inbuilt auth but through a web proxy could expose features through url's and acl's.

Proposed solution meerkat.example.com/view/ be setup that provides a view only listing of dashboards, eg: meerkat.example.com but without the edit, delete and clone options for dashboards plus removal of the Create New Dashboards and Edit Meerkat options.

This would allow a proxy to allow unauthorized or view authorized access to meerkat.example.com/view/* and authorized admin access to the url meerkat.example.com/.

**Comment by davekempe** [comment]
yeah the actual design should probably be move the current home page to /admin
and make the home page now just a view list
this will let the default be view only, and then we can hide /admin behind a reverse proxy basic auth until we do more surgery on meerkat

**Comment by ollytom** [comment]
Set a single username and password pair in configuration file on file system.

Backend: if auth set in config file, wrap relevant handlers for /dashboard paths in authentication middleware.

Frontend: In meerkat module, handle authentication challenge from meerkat. Include credentials option with `fetch()`.

**Comment by sol1-matt** [comment]
> yeah the actual design should probably be move the current home page to /admin and make the home page now just a view list this will let the default be view only, and then we can hide /admin behind a reverse proxy basic auth until we do more surgery on meerkat

the current paths are 
/ - default view
/edit/* - editing dashboards
/view/* - viewing dashboards

If you make / view only then you need to make /edit and /admin secure and anything added in the future.

If you assume everything is insecure and only /view is safe then there is less chance of messing up proxied auth, particularly with new meerkat features being in inadvertently exposed. 

TL;DR
safe
/ - secure
/view - insecure 

less safe
/ - insecure
/admin /edit - secure
